src/components/functions/utils/helpers.js

```diff
@@ -77,10 +77,21 @@ export function preparePoints(start, end, pointsCount, type) {
         pointsNo = pointsCount;
     }
 
+    let logStart = start;
+
+    if (logStart === 0) {
+        logStart = end / 10000;
+        alert("Logarithmic scale on X axis requested for invalid lower range value (zero).\n" +
```

Comment:
It's because it was triggered when creating series of point which is done after pressing `Submit` button. I change it to be displayed after switching axis.

Reply:
Now it is much better. But I'd also expect that after switching to log scale "Start" field would be automatically updated with new value (1e-4 * end).

Reply:
It can also be done a separate issue.